corehq/apps/cloudcare/static/cloudcare/js/form_entry/web_form_session.js

```diff
@@ -308,6 +308,13 @@ hqDefine("cloudcare/js/form_entry/web_form_session", [
             $.subscribe('formplayer.' + constants.CHANGE_LANG, function (e, lang) {
                 self.changeLang(lang);
             });
+            $.subscribe('formplayer.' + constants.DIRTY, function (e) {
+                hqRequire([
+                    "cloudcare/js/formplayer/app",
+                ], function (FormplayerFrontend) {
+                    FormplayerFrontend.trigger('setUnsavedFormInProgress');
```

Comment:
that's weird, as in it doesn't work if you defined `FormplayerFrontend` at the top of the file? But I've seen it imported in line like that elsewhere, so i'm guessing it's fine?

Reply:
> Yeah if I defined it at the top FormplayerFrontend was undefined.

This usually indicates that there's a circular dependency. I didn't trace it out, but `FormplayerFrontend` is prone to them, since it's something of a god object. This is indeed why `hqRequire` exists. Thanks for checking that the import at the top of the file didn't work.